providers/apache/beam/pyproject.toml

```diff
@@ -50,28 +50,31 @@ classifiers = [
     "Programming Language :: Python :: 3.12",
     "Topic :: System :: Monitoring",
 ]
-requires-python = "~=3.9"
+requires-python = "~=3.9,!=3.13"
 
 # The dependencies should be modified in place in the generated file.
 # Any change in the dependencies is preserved when the file is regenerated
 # Make sure to run ``breeze static-checks --type update-providers-dependencies --all-files``
 # After you modify the dependencies, and rebuild your Breeze CI image with ``breeze ci-image build``
 dependencies = [
     "apache-airflow>=2.10.0",
-    'apache-beam>=2.60.0',
-    "pyarrow>=14.0.1",
-    "numpy>=1.26.0",
-
+    'apache-beam>=2.60.0; python_version < "3.13"',
+    "pyarrow>=14.0.1; python_version < '3.13'",
+    "pyarrow>=18.0.0; python_version >= '3.13'",
+    "numpy>=1.22.4; python_version<'3.11'",
+    "numpy>=1.23.2; python_version=='3.11'",
+    "numpy>=1.26.0; python_version=='3.12'",
+    "numpy>=2.1.0; python_version>='3.13'",
```

Comment:
Does it make sense to define depednencies for Python 3.13 here if the provider is known not to be working in general? I'd propose to keep the dependencies as-is and define 3.13 changes once the provider works with Python 3.13 in general

Reply:
Yeah, it was being overcatious here. I think it will work without it.

Reply:
I will just leave upper-binding - just in case beam will not manage to update their deps for Python 3.14 as well.